File: 50-strukture/vezana-lista.js
class Clan {
  constructor(podatak) {
    this.podatak = podatak
    this.sledeci = null
  }
}

class Lista {
  constructor(vrednost) {
    this.glava = new Clan(vrednost)
  }

  dodaj(vrednost, trenutni = this.glava) {
    if (!trenutni) return
    if (!trenutni.sledeci) trenutni.sledeci = new Clan(vrednost)
    else this.dodaj(vrednost, trenutni.sledeci)
  }
}

const lista = new Lista('Mirko')
lista.dodaj('Slavko')
lista.dodaj('Fadil')
lista.dodaj('Španac')
console.log(JSON.stringify(lista, null, 4))

File: 50-strukture/vezana-lista.py

class Clan:
    def __init__(self, vrednost):
        self.vrednost = vrednost
        self.sledeci = None


class Lista:
    def __init__(self):
        self.glava = None

    def dodaj(self, vrednost):
        temp = Clan(vrednost)
        temp.sledeci = self.glava
        self.glava = temp

    def ispisi(self):
        trenutni = self.glava
        while trenutni:
            print(trenutni.vrednost)
            trenutni = trenutni.sledeci


lista = Lista()

lista.dodaj('Mirko')
lista.dodaj('Slavko')
lista.dodaj('Fadil')
lista.dodaj('Spanac')
lista.ispisi()
